File: app/src/main/java/forpdateam/ru/forpda/ui/fragments/devdb/device/comments/CommentsFragment.java
package forpdateam.ru.forpda.ui.fragments.devdb.device.comments;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import forpdateam.ru.forpda.App;
import forpdateam.ru.forpda.R;
import forpdateam.ru.forpda.ui.fragments.devdb.BrandFragment;
import forpdateam.ru.forpda.ui.fragments.devdb.device.SubDeviceFragment;

/**
 * Created by radiationx on 09.08.17.
 */

public class CommentsFragment extends SubDeviceFragment {
    private RecyclerView recyclerView;

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.device_fragment_specs, container, false);

        recyclerView = (RecyclerView) view.findViewById(R.id.base_list);
        recyclerView.setLayoutManager(new LinearLayoutManager(recyclerView.getContext()));
        CommentsAdapter adapter = new CommentsAdapter();
        adapter.addAll(device.getComments());
        recyclerView.setAdapter(adapter);
        recyclerView.addItemDecoration(new BrandFragment.SpacingItemDecoration(App.px8, true));
        return view;
    }
}

File: app/src/main/java/forpdateam/ru/forpda/ui/fragments/devdb/device/comments/CommentsFragment.kt
package forpdateam.ru.forpda.ui.fragments.devdb.device.comments

import android.os.Bundle
import android.view.View
import androidx.recyclerview.widget.LinearLayoutManager
import by.kirich1409.viewbindingdelegate.viewBinding
import forpdateam.ru.forpda.R
import forpdateam.ru.forpda.databinding.DeviceFragmentSpecsBinding
import forpdateam.ru.forpda.extensions.getDimenPx
import forpdateam.ru.forpda.ui.fragments.devdb.brand.DevicesFragment
import forpdateam.ru.forpda.ui.fragments.devdb.device.SubDeviceFragment

/**
 * Created by radiationx on 09.08.17.
 */

class CommentsFragment : SubDeviceFragment(R.layout.device_fragment_specs) {

    private val binding by viewBinding<DeviceFragmentSpecsBinding>()

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        val adapter = CommentsAdapter { item -> presenter.onCommentClick(item) }
        binding.baseList.layoutManager = LinearLayoutManager(binding.baseList.context)
        adapter.addAll(device.comments)
        binding.baseList.adapter = adapter
        binding.baseList.addItemDecoration(DevicesFragment.SpacingItemDecoration(binding.baseList.context.getDimenPx(R.dimen.dp8), true))
    }
}
